src/classes/frDonor.cls

```diff
@@ -64,90 +67,117 @@ public class frDonor extends frModel implements frSyncable{
     public Boolean sync() {
         Boolean result = false;
         Map<String, Object> request = getRequestBody();
-        supporterContact = null;
         String frId = getFunraiseId();
-        //try to find a donor that's already been integrated, use their funraise ID
-        List<Contact> contacts = (List<Contact>)Database.query('select Id, fr_ID__c, LastName from Contact where fr_ID__c = :frId');
-        if(contacts != null && contacts.size() > 0) {
-            supporterContact = contacts.get(0);
-        }
+    
+        SObject supporter = null; // Generic SObject for both Account and Contact
+        String sObjectType; // To track whether we're dealing with Account or Contact
         
+        Boolean isPersonAccount = frUtil.hasPersonAccountsEnabled();
+    
+        // Check if Person Accounts are enabled and set the appropriate SObject type
+        if (isPersonAccount) {
+            sObjectType = 'Account';
+        } else {
+            sObjectType = 'Contact';
+        }
+    
+        // Query based on fr_ID__c
+        List<SObject> records = Database.query(
+            'SELECT Id, fr_ID__c, LastName FROM ' + sObjectType + ' WHERE fr_ID__c = :frId'
+        );
+    
+        if (records != null && !records.isEmpty()) {
+            supporter = records.get(0);
+        }
+    
         String firstName = String.valueOf(request.get('firstName'));
         String lastName = String.valueOf(request.get('lastName'));
         String email = String.valueOf(request.get('email'));
-        if(supporterContact == null) {
-            if(String.isNotBlank(email)) {
-                //try to match on email + name, since not all orgs treat email as unique
-                if(String.isNotBlank(firstName) && String.isNotBlank(lastName)) {
-                    contacts = (List<Contact>)Database.query(
-                        'select Id, fr_ID__c, LastName from Contact where '+
-                        'Email = :email AND FirstName = :firstName AND LastName = :lastName '+
-                        'limit 1'
-                    );
-                    supporterContact = applyMatch(contacts, frId);
-                }
-                
-                //if name wasn't present or no match was found, try on email only
-                if(supporterContact == null) {
-                    contacts = (List<Contact>)Database.query('select Id, fr_ID__c, LastName from Contact where Email = :email limit 1');
-                    supporterContact = applyMatch(contacts, frId);                 
-                }
-                
+    
+        // Dynamic email field based on whether Person Accounts are enabled
+        String emailField = isPersonAccount ? 'PersonEmail' : 'Email';
+    
+        // Match based on email and names
+        if (supporter == null && String.isNotBlank(email)) {
```

Comment:
It looks like this changed the logic that was there previously.  Previously, the matching logic would:
Try to query on first name, last name, and email
If no match is found, then query just on email

This would allow the matching logic to get a more specific match (where names match) when there are multiple records with the same email.  This change has made it so that if the names and email don't match, then there won't be a match

Reply:
Added this back in the new update!